### PR TITLE
Optimizes dynamic lightning

### PR DESCRIPTION
**Pull request by Azarak**
## About The Pull Request
RANGE_TURFS is faster than typechecked turf in view

## Why It's Good For The Game
speed good

## Changelog
:cl:
code: optimizes dynamic lightning
/:cl:

### CODE REVIEW

File: code/datums/components/overlay_lighting.dm [hunk 1]
```diff
@@ -167,7 +167,9 @@
 	if(!current_holder)
 		return
 	var/atom/movable/light_source = GET_LIGHT_SOURCE
-	for(var/turf/lit_turf in view(lumcount_range, get_turf(light_source)))
+	var/turf/my_turf = get_turf(light_source)
+	for(var/t in RANGE_TURFS(lumcount_range, my_turf))
```
Comment (MrDoomBringer) [inline code/datums/components/overlay_lighting.dm:171]:
```suggestion
    //using RANGE_TURFS with untyped turfs is faster than using view() with typechecked turfs
	for(var/t in RANGE_TURFS(lumcount_range, my_turf))
```

Comment (MrDoomBringer) [reply code/datums/components/overlay_lighting.dm:171]: might be good to mention the reason for this for future developers

**Review by MrDoomBringer — CHANGES_REQUESTED** [review]

**Comment by YPOQ** [comment]
Won't this cause dynamic lights to cast lights through opaque objects/turfs?
Additionally it appears byond already can optimize this code.
https://gist.github.com/Giacom/be635398926bb463b42a#file-byond-performance-txt-L43

**Comment by Azarak** [comment]
> 
> 
> Won't this cause dynamic lights to cast lights through opaque objects/turfs?
> Additionally it appears byond already can optimize this code.
> https://gist.github.com/Giacom/be635398926bb463b42a#file-byond-performance-txt-L43

Oh I'm dumb, it will cause that issue

**Pull request closed without merging**